server/src/main/java/org/apache/druid/discovery/NodeRole.java

```diff
@@ -31,33 +38,88 @@
  * These abstractions can probably be merged when Druid updates to Jackson 2.9 that supports JsonAliases, see
  * see https://github.com/apache/druid/issues/7152.
  */
-public enum NodeRole
+public class NodeRole
```

Comment:
Please edit the javadoc:

- It's no longer an enum, the the javadoc says it is.
- The javadoc should explain the difference between "name" and "jsonName" and why they both exist.

Reply:
thanks, forgot to update, should be good now